### PR TITLE
feat(semver): Allow for passing includeBuild into format method

### PR DESCRIPTION
**Pull request by kamilogorek**
ref: https://github.com/denoland/deno_std/issues/3110

### CODE REVIEW

File: semver/mod.ts [hunk 1]
```diff
@@ -542,11 +542,14 @@ export class SemVer {
     this.format();
   }
 
-  format(): string {
+  format({ includeBuild } = { includeBuild: false }): string {
```
Comment (kt3k) [inline semver/mod.ts:545]: Should this be optional? How about showing the build metadata always?

Comment (kamilogorek) [reply semver/mod.ts:545]: We cannot do that, as `format()` call is used for semver comparison, ref: https://github.com/denoland/deno_std/issues/3110#issuecomment-1383221174

Also:
> Although format itself is not pure function, so maybe an alternative just for printing would be preferable.

**Comment by justinmchase** [comment]
It may be wise to just adjust the `.toString()` function to return the version with metadata and leave the format function alone, for backwards compatibility reasons. 

Also, it would be nice if there was a function which returned a new `SemVer` with the build set. since there is now way to apply build metadata to a version without doing string manipulation at the moment.

```ts
var v0 = semver.parse('1.2.3')
var v1 = v0.metadata('4')
console.log(v1.toString()) // 1.2.3+4
```

**Comment by justinmchase** [comment]
I made a more complete version of this PR with some feedback from the discussion:

https://github.com/denoland/deno_std/pull/3126

**Comment by kamilogorek** [comment]
Closed in favor of https://github.com/denoland/deno_std/pull/3126

**Pull request closed without merging**